tools/macos_buildenv.sh

```diff
@@ -90,26 +87,18 @@ case "$1" in
             rm "${SDKROOT}.tar.xz"
         fi
 
-        Qt5_DIR="$(find "${BUILDENV_PATH}" -type d -path "*/cmake/Qt5")"
-        [ -z "${Qt5_DIR}" ] && echo "Failed to locate Qt5_DIR!" >&2
-        QT_QPA_PLATFORM_PLUGIN_PATH="$(find "${BUILDENV_PATH}" -type d -path "*/plugins")"
-        [ -z "${QT_QPA_PLATFORM_PLUGIN_PATH}" ] && echo "Failed to locate QT_QPA_PLATFORM_PLUGIN_PATH" >&2
-        export CC="/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang"
-        export CXX="/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++"
-        export PATH="${BUILDENV_PATH}/bin:${PATH}"
-        export CMAKE_PREFIX_PATH="${BUILDENV_PATH}"
-        export Qt5_DIR
-        export QT_QPA_PLATFORM_PLUGIN_PATH
+        export VCPKG_ROOT="${BUILDENV_PATH}"
+        export VCPKG_OVERLAY_TRIPLETS="${BUILDENV_PATH}/overlay/triplets"
+        export VCPKG_DEFAULT_TRIPLET=x64-osx
+        export X_VCPKG_APPLOCAL_DEPS_INSTALL=ON
+        export CMAKE_GENERATOR=Ninja
 
         echo_exported_variables() {
-            echo "CC=${CC}"
-            echo "CXX=${CXX}"
```

Comment:
Bring this back. It is required for building locally to work in my testing.

Reply:
Please explain why.

Reply:
CMake doesn't find the C & C++ compilers without these variables set. I do not know why it works on GitHub Actions but not on my machine.

Reply:
Please share your cmake output.
I guess there must be something else wrong and this is only a workaround.

Reply:
This contains some related info:
https://stackoverflow.com/questions/24380456/how-can-i-make-cmake-use-gcc-instead-of-clang-on-mac-os-x

Reply:
That StackOverflow question is unrelated.

>  I guess there must be something else wrong and this is only a workaround.

So what? The solution was already there.

Reply:
Please post the output of plain 
./bootstrap-vcpkg.sh
In a fresh checked out vcpkg branch.

Reply:
Please understand it is quite a hassle for me to do anything on a Mac. I don't see any point continuing this thread. I don't care what the root cause is. The workaround was already in place. Just bring it back.

Reply:
Mandatory workarounds should be documented in the first place! Otherwise it is not unexpected if someone removes code that is apparently redundant or that is supposed to have no effect.

Reply:
@daschuer Please re-add the lines and amend the code with a comment about what we know so far about this workaround. Probably with a TODO that this code is supposed to be removed if possible.

To both of you: These back-and-forth discussions don't get us anywhere. Please avoid this in PRs.

Reply:
I think adding a comment explaining these environment variables are needed for CMake to find clang in local builds would be a good solution.

Reply:
ping